relayer/provider/cosmos/provider.go

```diff
@@ -888,16 +888,89 @@ func (cc *CosmosProvider) MsgTransfer(amount sdk.Coin, dstChainId, dstAddr, srcP
 // MsgRelayTimeout constructs the MsgTimeout which is to be sent to the sending chain.
 // The counterparty represents the receiving chain where the receipts would have been
 // stored.
-func (cc *CosmosProvider) MsgRelayTimeout(ctx context.Context, dst provider.ChainProvider, dsth int64, packet provider.RelayPacket, dstChanId, dstPortId, srcChanId, srcPortId string) (provider.RelayerMessage, error) {
+func (cc *CosmosProvider) MsgRelayTimeout(
+	ctx context.Context,
+	dst provider.ChainProvider,
+	dsth int64,
+	packet provider.RelayPacket,
+	dstChanId, dstPortId, srcChanId, srcPortId string,
+	order chantypes.Order,
+) (provider.RelayerMessage, error) {
 	var (
 		acc string
 		err error
+		msg provider.RelayerMessage
 	)
 	if acc, err = cc.Address(); err != nil {
 		return nil, err
 	}
 
+	switch order {
+	case chantypes.UNORDERED:
+		msg, err = cc.unorderedChannelTimeoutMsg(ctx, dst, dsth, packet, acc, dstChanId, dstPortId, srcChanId, srcPortId)
+		if err != nil {
+			return nil, err
+		}
+	case chantypes.ORDERED:
+		msg, err = cc.orderedChannelTimeoutMsg(ctx, dst, dsth, packet, acc, dstChanId, dstPortId, srcChanId, srcPortId)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("invalid order type %s, order should be %s or %s",
```

Comment:
This should be okay as a panic; if we have an invalid order value by this point, I assume it is a programmer error.